ui/src/services/alertUtils.js

```diff
@@ -0,0 +1,208 @@
+import type { PrometheusAlert, AlertList, Health } from './alertmanager/api';
+import { compareHealth } from './utils';
+
+// Return boolean if the two alerts have the same labels but different severity
+const isSameAlertWithDiffSeverity = (
+  alert1: {
+    [labelName: string]: string,
+    parents?: string[],
+    selectors?: string[],
+  },
+  alert2: {
+    [labelName: string]: string,
+    parents?: string[],
+    selectors?: string[],
+  },
+): boolean => {
+  // filter out the `severity` property
+  function replacer(key, value) {
+    if (key === 'severity') {
+      return undefined;
+    }
+    return value;
+  }
+
+  return JSON.stringify(alert1, replacer) === JSON.stringify(alert2, replacer);
+};
+
+/* 
+This function removes the warning alert from the list when critical one is triggered
+It should be called at saga level before storing the alerts to `redux-store`
+or where we resolve the promise with `react-query` 
+*/
+export const removeWarningAlerts = (alerts: AlertList): AlertList => {
+  const len = alerts.length;
+  const removeIndex = [];
+  for (let i = 0; i < len - 1; i++) {
+    for (let j = i + 1; j < len; j++) {
+      if (isSameAlertWithDiffSeverity(alerts[i].labels, alerts[j].labels)) {
+        if (alerts[i].labels.severity === STATUS_WARNING) {
+          removeIndex.push(i);
+        } else if (alerts[j].labels.severity === STATUS_WARNING) {
+          removeIndex.push(j);
+        }
+      }
+    }
+  }
+
+  let removedWarningAlerts = [...alerts];
+  removeIndex.forEach((index) => removedWarningAlerts.splice(index, 1));
+
+  return removedWarningAlerts;
+};
+
+// Sort the alerts base on the `severity`
+export const sortAlerts = (alerts: AlertList): AlertList => {
+  return alerts.sort(function (a, b) {
+    return compareHealth(a.labels.severity, b.labels.severity);
+  });
+};
+
+export const isObjectHasProperty = (obj: {}, property: string): boolean => {
+  return Object.prototype.hasOwnProperty.call(obj, property);
+};
+
+/* 
+Format the alerts
+If the label has multiple values, reshape it from comma-separated to an array
+Nets-alert:
+  node: 'node1'
+  selectors: ['node', '...']
+  parents: ['Node', '...']
+  comp: 'nets' 
+*/
+export const formatAlerts = (alerts: Array<PrometheusAlert>): AlertList => {
+  return alerts.map((alert) => {
+    let parents = [];
+    let selectors = [];
+
+    if (isObjectHasProperty(alert.labels, 'parents')) {
+      parents.push(alert.labels.parents.split(','));
+    }
+    if (isObjectHasProperty(alert.labels, 'selectors')) {
+      selectors.push(alert.labels.selectors.split(','));
+    }
+    return {
+      summary: alert.annotations.summary,
+      description: alert.annotations.description || alert.annotations.message,
+      startsAt: alert.startsAt,
+      endsAt: alert.endsAt || new Date().toISOString(),
+      severity: alert.labels.severity,
+      documentationUrl: alert.annotations.runbook_url,
+      labels: { ...alert.labels, parents, selectors },
+    };
```

Comment:
In order to avoid introducing a new condition + variable declaration every time we need a new array, I would purpose the following implementation :

```suggestion
    return {
      summary: alert.annotations.summary,
      description: alert.annotations.description || alert.annotations.message,
      startsAt: alert.startsAt,
      endsAt: alert.endsAt || new Date().toISOString(),
      severity: alert.labels.severity,
      documentationUrl: alert.annotations.runbook_url,
      labels: Object.fromEntries(Object.entries(alert.labels).map(([key, value]) => {
        if (key === 'parents' || key === 'selectors') {
           return [key, value.split(',')];
        }
        return [key, value];
      })),
    };
```

Reply:
Hi @JBWatenbergScality, 

`Object.fromEntries` requires node 12.0.0. I would say avoid using this method if it's possible.

> Running the platform UI locally
> Requirements
> Node.js, 10.16
https://metal-k8s.readthedocs.io/en/latest/developer/running/ui.html?highlight=UI%20locally#running-the-platform-ui-locally

Reply:
This only impacts tests execution which can import a polyfill to workarround the issue. Browser support of Object.fromEntries covers all the browser we want to support I believe : https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries#browser_compatibility. Also node v10 will be end of life by the end of April (https://nodejs.org/en/about/releases/)  I believe we should study an upgrade soon.